lib/modules/manager/gradle/readme.md

```diff
@@ -8,3 +8,8 @@ During [lock file maintenance](https://docs.renovatebot.com/configuration-option
 For regular dependency updates, renovate automatically updates lock state entries via the `--update-locks` command line flag.
 
 As the output of these commands can be very large, any text other than errors (in `stderr`) is discarded.
+
+### Dependency verification
+
+If Renovate finds a `gradle/verification-metadata.xml` file, it updates the content by using the `gradle --write-verification-metadata <hashTypes>` command.
+Renovate will check the file for existing hash types (like `sha256`) and use them as `<hashTypes>`.
```

Comment:
Looks like we only check `sha256`, `sha512` and `pgp` style hashes?

We should explain _which_ hash types Renovate can check and update. Or users will ask why their `md5` or `sha1` hash is not found/updated. 😄 

```ts
async function buildUpdateVerificationMetadataCmd(
  verificationMetadataFile: string | undefined,
  baseCmd: string
): Promise<string | undefined> {
  if (verificationMetadataFile) {
    const hashTypes: string[] = [];
    const verificationMetadata = await readLocalFile(verificationMetadataFile);
    for (const hashType of ['sha256', 'sha512']) {
      if (verificationMetadata?.includes(`<${hashType}`)) {
        hashTypes.push(hashType);
      }
    }
    if (!hashTypes.length) {
      hashTypes.push('sha256');
    }
    if (
      verificationMetadata?.includes(
        '<verify-signatures>true</verify-signatures>'
      )
    ) {
      hashTypes.push('pgp');
    }
    return `${baseCmd} --write-verification-metadata ${hashTypes.join(
      ','
    )} help`;
  }
}
```

Reply:
I'd suggest adding a note that hash functions that gradle also supports verification metadata with `md5` and `sha1` but renovate doesn't take those into account since they are prone to collision attacks. In case hashes are encountered that were generated with any of these algorithms, renovate defaults to `sha256`.

Reply:
@Churro Good idea! Please review this suggestion:

```suggestion
Renovate will check the file for existing hash types (like `sha256`) and use them as `<hashTypes>`.

<!-- prettier-ignore -->
!!! warning
    Gradle allows verification metadata to use the `md5` and `sha1` algorithms.
    Because those algorithms are prone to collision attacks, Renovate ignores them.
    If Renovate encounters hashes that are generated with `md5` or `sha1` algorithms, Renovate uses `sha256` instead.
```